diff -ruN dash-0.16.1.1-fix-bind-namespace-errors/src/qt/trafficgraphwidget.cpp dash-0.16.1.1-fix-QPainterPath-issue/src/qt/trafficgraphwidget.cpp
--- dash-0.16.1.1-fix-bind-namespace-errors/src/qt/trafficgraphwidget.cpp	2021-01-19 18:55:19.474888101 -0500
+++ dash-0.16.1.1-fix-QPainterPath-issue/src/qt/trafficgraphwidget.cpp	2021-01-19 22:21:00.017139488 -0500
@@ -8,6 +8,7 @@
 
 #include <boost/bind/bind.hpp>
 
+#include <QPainterPath>
 #include <QPainter>
 #include <QColor>
 #include <QTimer>
